test-utils/test-runner/src/lib.rs

```diff
@@ -0,0 +1,409 @@
+//! 		(
+//! 			Arc<TFullClient<Self::Block, Self::RuntimeApi, Self::Executor>>,
+//! 			Arc<TFullBackend<Self::Block>>,
+//! 			KeyStorePtr,
+//! 			TaskManager,
+//! 			InherentDataProviders,
+//! 			Option<Box<
+//! 				dyn ConsensusDataProvider<
+//! 					Self::Block,
+//! 					Transaction = TransactionFor<
+//! 						TFullClient<Self::Block, Self::RuntimeApi, Self::Executor>,
+//! 						Self::Block
+//! 					>,
+//! 				>
+//! 			>>,
+//! 			Self::SelectChain,
+//! 			Self::BlockImport
+//! 		),
+//! 		sc_service::Error
+//! 	> {
+//! 		let (
+//! 			client,
+//! 			backend,
+//! 			keystore,
+//! 			task_manager,
+//! 		) = new_full_parts::<Self::Block, Self::RuntimeApi, Self::Executor>(config)?;
+//! 		let client = Arc::new(client);
+//!
+//! 		let inherent_providers = InherentDataProviders::new();
+//! 		let select_chain = sc_consensus::LongestChain::new(backend.clone());
+//!
+//! 		let (grandpa_block_import, ..) =
+//! 			sc_finality_grandpa::block_import(client.clone(), &(client.clone() as Arc<_>), select_chain.clone())?;
+//!
+//! 		let (block_import, babe_link) = sc_consensus_babe::block_import(
+//! 			sc_consensus_babe::Config::get_or_compute(&*client)?,
+//! 			grandpa_block_import,
+//! 			client.clone(),
+//! 		)?;
+//!
+//! 		let consensus_data_provider = BabeConsensusDataProvider::new(
+//! 			client.clone(),
+//! 			keystore.clone(),
+//! 			&inherent_providers,
+//! 			babe_link.epoch_changes().clone(),
+//! 			vec![(AuthorityId::from(Alice.public()), 1000)]
+//! 		)
+//! 		.expect("failed to create ConsensusDataProvider");
+//!
+//! 		Ok((
+//! 			client,
+//! 			backend,
+//! 			keystore,
+//! 			task_manager,
+//! 			inherent_providers,
+//! 			Some(Box::new(consensus_data_provider)),
+//! 			select_chain,
+//! 			block_import
+//! 		))
+//! 	}
+//!
+//! 	fn dispatch_with_root(call: <Self::Runtime as frame_system::Config>::Call, node: &mut Node<Self>) {
+//!         let alice = MultiSigner::from(Alice.public()).into_account();
+//!         let call = pallet_sudo::Call::sudo(Box::new(call)); // :D
+//!         node.submit_extrinsic(call, alice);
+//!         node.seal_blocks(1);
+//!     }
+//! }
+//!
+//! /// And now for the most basic test
+//!
+//! #[test]
+//! fn simple_balances_test() {
+//! 	// given
+//! 	let mut node = Node::<Requirements>::new().unwrap();
+//!
+//! 	type Balances = pallet_balances::Module<node_runtime::Runtime>;
+//!
+//! 	let (alice, bob) = (Alice.pair(), Bob.pair());
+//! 	let (alice_account_id, bob_acount_id) = (
+//!         MultiSigner::from(alice.public()).into_account(),
+//!         MultiSigner::from(bob.public()).into_account()
+//!     );
+//!
+//!     /// the function with_state allows us to read state, pretty cool right? :D
+//! 	let old_balance = node.with_state(|| Balances::free_balance(alice_account_id.clone()));
+//!
+//!     // 70 dots
+//!     let amount = 70_000_000_000_000;
+//!
+//!     /// Send extrinsic in action.
```

Comment:
these lines are not indented properly, and I think the line width is not correct. Would be good if you wrap them at the 100 cols, which is the substrate standard.